packages/strapi/lib/core/configurations.js

```diff
@@ -318,9 +318,30 @@ module.exports.app = async function() {
     return acc;
   }, {});
 
+  // default settings
   this.config.port = get(this.config.currentEnvironment, 'server.port') || this.config.port;
   this.config.host = get(this.config.currentEnvironment, 'server.host') || this.config.host;
-  this.config.url = `http://${this.config.host}:${this.config.port}`;
+
+  // default construct url
+  this.config.url = `http://${this.config.host}:${this.config.port}`
+
+  // proxy settings
+  this.config.proxy = this.config.currentEnvironment.server.proxy;
+  this.config.proxy.enabled = get(this.config.currentEnvironment, 'server.proxy.enabled') || this.config.proxy.enabled;
+  this.config.proxy.port = get(this.config.currentEnvironment, 'server.proxy.port') || this.config.proxy.port;
+  this.config.proxy.host = get(this.config.currentEnvironment, 'server.proxy.host') || this.config.proxy.host;
+  this.config.proxy.ssl = get(this.config.currentEnvironment, 'server.proxy.ssl') || this.config.proxy.ssl;
+
+  // check if SSL enabled and construct url
+  function getProxyUrl(ssl, url) {
+  if (ssl) return `https://${url}`;
```

Comment:
Warning with indents, there is an incorrect indentation in this function. 
I'm surprise because there is a **lint** on `beforeCommit` and it have to catch that...

And you can have something like that:
```js
function getProxyUrl(ssl, url) {
  return `http${ssl ? 's' : ''}://${url}`;
}
```